File: cpp/283_移动零_easy.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int N = nums.size();
        int left=0, right=0;
        while(right<N)
        {
            if(nums[right])
            {
                swap(nums[left], nums[right]);
                left++;
            }
            right++;
        }
    }
};

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int N = nums.size();
        int index = 0;
        for(int i=0;i<N;i++)
            if(nums[i])
                nums[index++]=nums[i];
        for(int i=index;i<N;i++)
            nums[i]=0;
    }
};

File: python/283_移动零_easy.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        N = len(nums)
        left, right = 0, 0
        while(right<N):
            if nums[right]:
                temp = nums[left]
                nums[left]=nums[right]
                nums[right]=temp
                left += 1
            right += 1

class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        N = len(nums)
        index = 0
        for i in range(N):
            if nums[i]:
                nums[index] = nums[i]
                index += 1
        for i in range(index, N):
            nums[i] = 0